### PR TITLE
Feature/organization queries

### PR DESCRIPTION
**Pull request by Caleb-Sutherland**
Added queries and types for organization service.

### CODE REVIEW

**Comment by CaioCamatta** [comment]
Take a brief look at the failing tests

File: src/controllers/organization.controller.ts [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
 class OrganizationController {
   public organizationService = new OrganizationService();
 
   public getOrganizationData = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
-      const orgId = parseInt(req.params.id);
-      const orgData = await this.organizationService.getOrganizationData(orgId);
+      const orgId = req.params.id;
+      const queryObject = url.parse(req.url, true).query;
```
Comment (CaioCamatta) [inline src/controllers/organization.controller.ts:11]: What does this do? `url.parse(req.url, true).query;`. Do we need to use the `url` library?

Comment (Caleb-Sutherland) [reply src/controllers/organization.controller.ts:11]: Its just a way I found to extract query parameters

Comment (Caleb-Sutherland) [reply src/controllers/organization.controller.ts]: Changed it

Comment (Caleb-Sutherland) [reply src/controllers/organization.controller.ts]: Also checked out the failing tests and it was stuff that I did not touch, but might've broke when Jack changed the database - anyways, it is fixed now

File: src/app.ts [hunk 2]
```diff
@@ -46,7 +46,10 @@ class App {
 
   private initializeMiddlewares() {
     this.app.use(morgan(config.get('log.format'), { stream }));
-    // this.app.use(cors({ origin: config.get('cors.origin'), credentials: config.get('cors.credentials') }));
+
+    //const corsOptions = { origin: config.get('cors.origin'), credentials: config.get('cors.credentials') };
+    const corsOptions: CorsOptions = { origin: '*' };
+    this.app.use(cors(corsOptions));
```
Comment (CaioCamatta) [inline src/app.ts:52]: Just a note for myself to address CORS in my next PR

File: src/controllers/organization.controller.ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
 import OrganizationService from '@/services/organization.service';
 import { NextFunction, Request, Response } from 'express';
+import { start } from 'repl';
```
Comment (CaioCamatta) [inline src/controllers/organization.controller.ts:3]: Are we using this?

File: src/controllers/organization.controller.ts [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
-      const orgData = await this.organizationService.getOrganizationData(orgId);
+      const orgId = req.params.id;
 
+      if (!req.query.start_date || !req.query.end_date) {
+        throw new HttpException(400, 'Please pass start and end date parameters.');
+      }
+
+      const start_date = req.query.start_date.toString();
+      const end_date = req.query.end_date.toString();
+      const orgData = await this.organizationService.getOrganizationData(orgId, start_date, end_date);
```
Comment (CaioCamatta) [inline src/controllers/organization.controller.ts:19]: Use camel case for consistency. 
I used snake case in the `interface.ts` files just to match the database but should've probably used camel case there too

File: src/services/organization.service.ts [hunk 5]
```diff
@@ -15,8 +24,28 @@ class OrganizationService {
       throw new HttpException(404, 'Organization not found.');
     }
 
+    // Setup date objects to be used for query (start/end date for the requested period)
+    const start_date = new Date(startDate);
+    const end_date = new Date(endDate);
```
Comment (CaioCamatta) [inline src/services/organization.service.ts:29]: Use camel case

**Pull request merged**